functional/util/util.go

```diff
@@ -154,3 +154,32 @@ func WaitForState(stateCheckFunc func() bool) (time.Duration, error) {
 		}
 	}
 }
+
+// SliceContains returns true if every string in needle slice exists in any
+// slice in haystack.
+func SliceContains(haystack []string, needle []string) bool {
```

Comment:
You should probably check that the list of units is *identical* to the expected ones, not that it's a superset...

But for the actual test cases in question, it's probably overkill anyway -- see my first comment, about just checking the count...